### ISSUE TITLE
hide.NS option for add_pvalue function

### ISSUE THREAD

**Issue by jaydoc** [csdaw/ggprism#7]
This is an option in stat_pvalue_manual to hide insignificant p values from being plotted and seems to be missing from add_pvalue function. Would you be able to add it?

**Comment by csdaw** [comment]
Hi there. I left out this option on purpose as it assumed that your significance level would always be 0.05, which might not always be the case.

I felt it was better to just filter out the non-significant rows from the p-value data.frame before plotting, as below:

``` r
library(ggplot2)
library(ggprism)
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union

df_p_val <- ToothGrowth %>%
  rstatix::group_by(dose) %>%
  rstatix::t_test(len ~ supp) %>%
  rstatix::adjust_pvalue(p.col = "p", method = "bonferroni") %>%
  rstatix::add_significance(p.col = "p.adj") %>% 
  rstatix::add_xy_position(x = "dose", dodge = 0.8) %>%
  filter(p.adj < 0.01)          # here we have chosen an alpha of 0.01

p <- ggplot(ToothGrowth, aes(x = factor(dose), y = len)) + 
  geom_boxplot(aes(fill = supp)) + 
  theme_prism() + 
  coord_cartesian(ylim = c(0, 40))

p + add_pvalue(df_p_val, 
               xmin = "xmin", 
               xmax = "xmax",
               label = "p = {p.adj}",
               tip.length = 0)
```

![](https://i.imgur.com/SgFc2B5.png)

<sup>Created on 2021-04-05 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Comment by jaydoc** [comment]
I can see why you left it out. 

In my use case, I am using add_pvalue as part of a function in which I am providing custom `y.position` values as a vector of values calculated on-the-fly when the function is called.

If I filter the p-values it creates errors regarding vector lengths and `'list' object cannot be coerced to type 'double'`.

I will try to provide a MWE if required.

**Comment by csdaw** [comment]
Yes a MWE would be helpful :) I expect there is a mismatch between the length of `y.position` and the nrow of`data`.

As a side note, looking at the [source](https://github.com/kassambara/ggpubr/blob/ac5a01f586a5f0b47260fed071e79be36ef3f8ea/R/stat_pvalue_manual.R#L359-L376`) for `stat_pvalue_manual()`, the hide.ns option calls the internal `remove_ns()` function which just uses `dplyr::filter()` as well so it shouldn't function any differently to just filtering beforehand.

**Comment by dannychu1108** [comment]
Hi! I'm trying to use your method to filter out the ns values, but I received an error:
Error:
! Problem while computing aesthetics.
ℹ Error occurred in the 4th layer.
Caused by error in `check_aesthetics()`:
! Aesthetics must be either length 1 or the same as the data (1)
✖ Fix the following mappings: `fill`
